minoc/src/intTest/java/org/opencadc/minoc/SetStateTest.java

```diff
@@ -0,0 +1,397 @@
+import ca.nrc.cadc.net.HttpPost;
+import ca.nrc.cadc.net.HttpTransfer;
+import ca.nrc.cadc.net.HttpUpload;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import ca.nrc.cadc.rest.RestAction;
+import ca.nrc.cadc.util.Log4jInit;
+
+import static org.junit.Assert.fail;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URL;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.security.auth.Subject;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test PUT, GET, DELETE, POST, HEAD and common errors
+ * Note: This test the minoc-test.pem cert and requires minoc-availability.properties 
+ *       to be in minoc/config/. The following entry needs to be in the 
+ *       minoc-availability.properties file.
+ *       users=CN=cadcauthtest1_24c, OU=cadc, O=hia, C=ca
+ * 
+ * @author yeunga
+ */
+public class SetStateTest extends MinocTest {
+    
+    private static final Logger log = Logger.getLogger(SetStateTest.class);
+    
+    static {
+        Log4jInit.setLevel("org.opencadc.minoc", Level.INFO);
+    }
+    
+    public SetStateTest() {
+        super();
+    }
+    
+    @Test
+    public void testSetOffline() throws Exception {
+        try {
+            // set minoc state to 'Offline'
+            String state = RestAction.STATE_OFFLINE;
+	    setState(state);
+
+            // test put, post, get, head, delete actions
+            URI artifactURI = URI.create("cadc:TEST/file.txt");
+            URL artifactURL = new URL(filesURL + "/" + artifactURI.toString());
+
+            String content = "abcdefghijklmnopqrstuvwxyz";
+            String encoding = "test-encoding";
+            String type = "text/plain";
+            byte[] data = content.getBytes();
+
+            // put: no length or checksum
+            InputStream in = new ByteArrayInputStream(data);
+            HttpUpload put = new HttpUpload(in, artifactURL);
+            put.setRequestProperty(HttpTransfer.CONTENT_TYPE, type);
+            put.setRequestProperty(HttpTransfer.CONTENT_ENCODING, encoding);
+            put.setDigest(computeChecksumURI(data));
+
+            Subject.doAs(userSubject, new RunnableAction(put));
+            Assert.assertNotNull("put to minoc in offline state should have failed", put.getThrowable());
+            String exMsg = put.getThrowable().getMessage();
+            Assert.assertTrue("incorrect exception message from put", exMsg.contains("System is offline"));
+
+            // get
+            OutputStream out = new ByteArrayOutputStream();
+            HttpGet get = new HttpGet(artifactURL, out);
+            Subject.doAs(userSubject, new RunnableAction(get));
```

Comment:
before asserts, it is useful to log (info) the response code and throwable... helps when figuring out why a test stopped passing. 

There should be an assert for the expected response code (before checking the throwable because if the code is wrong then something else happened and wrong code is more important). Once you add the above log message you can figure out what code you get now, add that assert, and then we'll decide if that's actually right.

Reply:
Added log (info) of response code and throwable. Added an assert for the expected response code.